Userland/Utilities/grep.cpp

```diff
@@ -168,12 +168,12 @@ int main(int argc, char** argv)
             return true;
         };
 
-        auto add_directory = [&handle_file](String base, Optional<String> recursive, auto handle_directory) -> void {
+        auto add_directory = [&handle_file, &files](String base, Optional<String> recursive, auto handle_directory) -> void {
```

Comment:
Let's not capture the entirety of `files` here, how about a `auto user_has_specified_files = !files.is_empty();` and capturing this variable (and using it below too)?

Reply:
done!